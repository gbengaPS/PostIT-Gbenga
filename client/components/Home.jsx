import React from 'react';
import { Link } from 'react-router-dom';
import Nav from './common/Nav';

/**
 *
 * @class Home
 * @extends {React.Component}
 */
class Home extends React.Component {
  /**
   *
   * @returns {void}
   * @memberof Home
   */
  componentWillUpdate() {
    $('.button-collapse').sideNav();
  }
  /**
   *
   *
   * @returns {object} -jsx
   * @memberof Home
   */
  render() {
    const rightLinkObject = (
      <div>
        <ul className="hide-on-med-and-down">
          <li><Link className="" to="/login" > Login </Link> </li>
          <li><Link className="" to="/signup" > Signup </Link> </li>
        </ul>
          <ul id="slide-out" className="side-nav">
            <p />
            <li><Link to="/login">Login</Link></li>
            <li><Link to="/signup">Signup</Link></li>
          </ul>
        <Link
          to="#"
          data-activates="slide-out"
          className="right button-collapse">
          <i className="material-icons">menu</i>
        </Link>
      </div>
    );
    return (
      <div >
        <Nav
        rightLink={rightLinkObject}
        />
        <div className="centralized-div row">
          <div className="col m4 offset-m3 s10 offset-s1" >
          <h1 className="bold-font"> Built for</h1>
            <h2 className="bold-font">Group messaging</h2>
            <p className="postit-text">
              Postit allows you communicate within the scope of a group.
              You get to create groups, join existing
              ones and send prioritized
              messages to group members.
              </p>
            <Link to="signup" className= "center btn light-blue darken-4">
              Get Started
            </Link>
            <p className="clearfix" />
          </div>
        </div>
      </div>
    );
  }
}
export default Home;
