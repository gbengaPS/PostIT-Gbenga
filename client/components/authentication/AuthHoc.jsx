import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import UserNav from '../common/UserNav';
import AppNav from '../common/AppNav';

  /**
   * @description Authentication HOC
   *
   * @param {jsx} Component -jsx component
   *
   * @returns {jsx} -return component
  */
export default (Component) => {
  let user, rightLinkObject;

  /**
   *
   * @class AuthHoc
   *
   * @extends {React.Component}
   */
  class AuthHoc extends React.Component {
    /**
     *
     * @returns {void} -returns nothing
     */
    componentWillMount() {
      if (!localStorage.getItem('postitToken')) {
        this.props.history.push('/login');
      }
    }

    /**
     *
     * @return {void} -returns nothing
     *
     * @memberof AuthHoc
     */
    componentDidMount() {
      $('.dropdown-button').dropdown();
      $('.collapsible').collapsible();
      $('.button-collapse').sideNav();
    }
    /**
     * @description render function
     *
     * @returns {jsx} -jsx representation of the component
     */
    render() {
      user = JSON.parse(localStorage.getItem('postitUser'));
      rightLinkObject = (
        <Link
        className='dropdown-button big'
        to='#'
        id="username"
        data-activates='userDropdown'
        >
          {user.username}&nbsp;
          <i className="fa fa-caret-down" ></i>
        </Link>
      );
      return (
        <div>
          <UserNav
          rightLink={rightLinkObject}
          history={this.props.history}
          />
          <div className="component-container row">
          <AppNav />
          <Component {...this.props} />
          </div>
        </div>
      );
    }
  }
  AuthHoc.propTypes = {
    history: PropTypes.object.isRequired,
  };
  /**
   * @description Maps state to props
   *
   * @param {object} state -application state
   *
   * @returns {object} -returns part of the state
   */
  const mapStateToProps = state => (
    {
      user: state.authReducer.user,
    }
  );
  return connect(mapStateToProps)(AuthHoc);
};
