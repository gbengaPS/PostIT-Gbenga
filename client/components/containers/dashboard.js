import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter, Link } from 'react-router-dom';
import { getGroups } from '../../actions/groupAction';
import UserDashboard from '../presentational/userDashboard.jsx';

class Dashboard extends React.Component {
  componentWillMount() {

  }
  componentDidMount() {
    this.props.getGroups();
    console.log(this.props);
  }
  render() {
    let groups;
    let userGroups;
    if (this.props.groups.groups) {
      console.log(this.props.user);
      groups = this.props.groups.groups;
      console.log('groups', this.props.groups.groups);
      userGroups = (
        <ul>
          {groups.map(group => (
            <li key={group.id}><Link to={`/group/${group.id}`}> {group.groupName} </Link> </li>
            ))}
        </ul>);
    }
    return (
      <div>
        <UserDashboard userGroups={userGroups} />
      </div>
    );
  }
}

Dashboard.propTypes = {
  userGroupSuccess: PropTypes.bool.isRequired,
  // groups: PropTypes.object.isRequired,
};

const mapStateToProps = state => (
  {
    groups: state.userGroupReducer,
    userGroupSuccess: state.getUserGroupSuccess,
    user: state.userAuth,
    error: state.loginError,
  }
);
const mapDispatchToProps = (dispatch) => {
  return {
    getGroups: () => {
      dispatch(getGroups());
    },
  };
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Dashboard));