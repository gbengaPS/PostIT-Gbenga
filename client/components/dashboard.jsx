import React from 'react';

class Dashboard extends React.Component {
  componentDidMount(){
    console.log(this.props)
  }
  render() {
    return (
      <div>
        <p>Welcome to the dashboard</p>
      </div>
      );
  }
}
export default Dashboard;
