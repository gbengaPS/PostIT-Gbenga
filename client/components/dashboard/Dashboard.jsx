import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';

/**
 * @description Dashboard class
 *
 * @class Dashboard
 *
 * @extends {React.Component}
 */
export class Dashboard extends React.Component {

  /**
   * @description render fucntion
   *
   * @returns {jsx} -jsx representation of the component
   */
  render() {
    let groups;
    let userGroups;
    if (this.props.groups.length > 0) {
      groups = this.props.groups;
      userGroups = (
        groups.map(group => (
          <div className="postit-card col m3 s5" key={group.id}>
          <p className="header">
            <Link className="" to={`/group/${group.id}`}>
            {group.groupName}
            </Link>
          </p>
          <p>{group.groupDescription} </p>
          <p className="bottom-element">
            Active: <span className="bold">
            {group.messages[0] ?
              moment(group.messages[0].updatedAt).fromNow() :
              'Not yet active' }
            </span>
          </p>
        </div>
        ))
      );
    }
    return (
      <div>
        <div
          className="col m7 offset-m1 s12 component-container groups-display">
          <p className="center header">My Groups</p>
          <div className="divider"/>
          {userGroups}
        </div>
      </div>
    );
  }
}

Dashboard.propTypes = {
  groups: PropTypes.array.isRequired,
};

/**
 * @description Maps state to props
 *
 * @param {object} state -application state
 *
 * @returns {object} -returns part of the state
*/
const mapStateToProps = state => (
  {
    groups: state.groupReducer.groups,
  }
);

export default connect(mapStateToProps, null)(Dashboard);
