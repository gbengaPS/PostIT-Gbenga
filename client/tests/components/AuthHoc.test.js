import React from 'react';
import AuthHoc from '../../components/authentication/AuthHoc';
import Home from '../../components/Home';
import { Dashboard } from '../../components/dashboard/Dashboard';

jest.mock('../../components/authentication/AuthHoc');
describe('AuthHoc component', () => {
  it('should always return the componet passed to it', () => {
    console.log(AuthHoc(<Home />));
    const wrappedComponent = mount(AuthHoc(<Home />));
    expect(wrappedComponent).toEqual(shallow(<Home />));
  });
  it('should should render correctly', () => {
    const Component = AuthHoc(Dashboard);
    const wrapper = shallow(<Component />);
    console.log(wrapper);
    expect(wrapper).toMatchSnapshot();
  });
});
