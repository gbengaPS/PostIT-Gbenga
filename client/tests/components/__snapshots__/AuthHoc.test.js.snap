// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHoc component should should render correctly 1`] = `undefined`;
